contentcuration/contentcuration/frontend/channelList/views/Channel/StudioMyChannels.vue

```diff
@@ -0,0 +1,582 @@
+          this.removeViewer({ channelId: this.selectedChannel.id, userId: currentUserId }).then(
+            () => {
+              this.deleteDialog = false;
+              this.$store.dispatch('showSnackbarSimple', this.$tr('channelRemovedSnackbar'));
+            },
+          );
+        } else {
+          this.deleteChannel(this.selectedChannel.id).then(() => {
+            this.deleteDialog = false;
+            this.$store.dispatch('showSnackbarSimple', this.$tr('channelDeletedSnackbar'));
+          });
+        }
+      },
+      thumbnailSrc(channel) {
+        return channel.thumbnail_encoding && channel.thumbnail_encoding.base64
+          ? channel.thumbnail_encoding.base64
+          : channel.thumbnail_url;
+      },
+      getPublishStatus(channel) {
+        return channel.last_published
+          ? this.$tr('lastPublished', {
+            last_published: this.$formatRelative(channel.last_published, {
+              now: new Date(),
+            }),
+          })
+          : this.$tr('unpublishedText');
+      },
+    },
+    $trs: {
+      channel: 'New channel',
+      noChannelsFound: 'No channels found',
+      resourceCount: '{count, plural,\n =1 {# resource}\n other {# resources}}',
+      unpublishedText: 'Unpublished',
+      lastPublished: 'Published {last_published}',
+      lastUpdated: 'Updated {updated}',
+      details: 'Details',
+      viewContent: 'View channel on Kolibri',
+      goToWebsite: 'Go to source website',
+      editChannel: 'Edit channel details',
+      copyToken: 'Copy channel token',
+      deleteChannel: 'Delete channel',
+      deleteTitle: 'Delete this channel',
+      removeChannel: 'Remove from channel list',
+      removeBtn: 'Remove',
+      removeTitle: 'Remove from channel list',
+      deletePrompt: 'This channel will be permanently deleted. This cannot be undone.',
+      removePrompt:
+        'You have view-only access to this channel. Confirm that you want to remove it from your list of channels.',
+      channelDeletedSnackbar: 'Channel deleted',
+      channelRemovedSnackbar: 'Channel removed',
+      channelLanguageNotSetIndicator: 'No language set',
+      cancel: 'Cancel',
+    },
+  };
+
+</script>
+
+
+<style lang="scss" scoped>
+
+  .window-small {
+    .channels-body,
+    .new-channel {
+      width: 100%;
+    }
+  }
+
+  .window-medium {
+    .channels-body,
+    .new-channel {
+      width: 100%;
+      max-width: 83.33%;
+    }
+  }
+
+  .window-large {
+    .channels-body,
+    .new-channel {
+      width: 100%;
+      max-width: 50%;
+    }
+  }
+
+  .windowBreakpoint-3 {
+    .channels-body,
+    .new-channel {
+      width: 100%;
+      max-width: 83.33%;
+    }
+  }
+
+  .windowBreakpoint-4 {
+    .channels-body,
+    .new-channel {
+      width: 100%;
+      max-width: 66.66%;
+    }
+  }
+
+  .windowBreakpoint-5 {
+    .channels-body,
+    .new-channel {
+      width: 100%;
+      max-width: 50%;
+    }
+  }
```

Comment:
This styling is somewhat repetitive - it looks like in every case the width is set to 100%. Also, it's not entirely clear to me which class will take precedence, the windowBreakpoint-N class or the window size class.

```suggestion
  .channels-body,
  .new-channel {
    width: 100%;
  }
```

The breakpoint specific logic might then be better suited to a dynamic style using the following computed prop:

```
maxWidth() {
   if (this.windowBreakpoint < 2) {
      return null;
   }
   if (this.windowBreakpoint === 3) {
      return '83.33%';
   }
.....
```

and then apply it to the base class - as we are already conditionally applying the classes based on the breakpoints, we may as well just conditionally apply the single style that those classes conditionalize.

Reply:
Sure, will update this